### PR TITLE
Prime Number Testing

### PR DESCRIPTION
**Pull request by ejohnstown**
Made modifications to the primality testing so that the Miller-Rabin tests check against up to 40 random numbers rather than a fixed list of small primes.

### CODE REVIEW

**Comment by ejohnstown** [comment]
Do not accept this pull request.

**Comment by ejohnstown** [comment]
jenkins retest this please

File: src/ssl.c [hunk 1]
```diff
@@ -22538,13 +22545,41 @@ int wolfSSL_BN_is_prime_ex(const WOLFSSL_BIGNUM *bn, int nbchecks,
...
+                                 nbchecks, &res, rng) != MP_OKAY) {
+            WOLFSSL_MSG("mp_prime_is_prime_ex error");
+            res = MP_NO;
+        }
+    }
+
+    if (initTmpRng)
+        wc_FreeRng(tmpRNG);
+#ifdef WOLFSSL_SMALL_STACK
+    tmpRNG = (WC_RNG*)XMALLOC(sizeof(WC_RNG), NULL, DYNAMIC_TYPE_RNG);
```
Comment (dgarske) [inline src/ssl.c]: I think this line shouldn't be here? It was XMALLOC'd above. Leak... Makes me wonder if Jenkins is running valgrind with small stack? @kaleb-himes can you check that?

File: wolfcrypt/src/tfm.c [hunk 2]
```diff
@@ -2940,15 +2911,144 @@ int fp_isprime_ex(fp_int *a, int t)
...
+}
+
+#endif /* NO_RSA NO_DSA NO_DH WOLFSSL_KEY_GEN */
+
+
+#ifdef WOLFSSL_KEY_GEN
+
+static void fp_gcd(fp_int *a, fp_int *b, fp_int *c);
+static void fp_lcm(fp_int *a, fp_int *b, fp_int *c);
+static int  fp_randprime(fp_int* N, int len, WC_RNG* rng, void* heap);
```
Comment (dgarske) [inline wolfcrypt/src/tfm.c]: Do you think it might be cleaner to have these declarations in tfm.h and not be static?

File: wolfcrypt/test/test.c [hunk 3]
```diff
@@ -19131,6 +19141,189 @@ int mp_test(void)
...
+} pairs_t;
+
+
+/*
+n =p1p2p3, where pi = ki(p1−1)+1 with (k2,k3) = (173,293)
+p1 = 2^192 * 0x000000000000e24fd4f6d6363200bf2323ec46285cac1d3a
+   + 2^0   * 0x0b2488b0c29d96c5e67f8bec15b54b189ae5636efe89b45b
+*/
+
+const unsigned char c192a[] =
```
Comment (dgarske) [inline wolfcrypt/test/test.c]: Can you make these static? I can't see a reason to make them exportable.

File: wolfcrypt/src/tfm.c (hunk 2 above)
Comment (ejohnstown) [reply wolfcrypt/src/tfm.c]: I don't think it matters one way or the other. Really, I think they should all be static and we don't have a tfm.h. tfm.c has mp_* wrappers for all the fp_* functions.

File: src/ssl.c (hunk 1 above)
Comment (ejohnstown) [reply src/ssl.c]: Stupid copy-paste error. Thanks.

**Comment by toddouska** [comment]
thanks!

**Pull request merged**